backend/routes/generate.js

```diff
@@ -0,0 +1,290 @@
+var express = require('express');
+const { stack } = require('./db.js');
+var router = express.Router();
+var database = require('./db.js');
+var classData = database.classData;
+
+var conflictPairs = [];
+
+router.post('/', function(req, res) {
+    conflictPairs = [];
+    var generateRequest = req.body;
+    //console.log(generateRequest);
+    var generateResult = generateSchedules(generateRequest);
+    // console.log(generateResult);
+    res.send(generateResult);
+});
+
+function unitConstraint(possibleSchedules, minUnits, maxUnits){
+    // for (var schedule of possibleSchedules) {
+    for (var i=0; i<possibleSchedules.length; i++) {
+        schedule = possibleSchedules[i]
+        unitSum = 0;
+        // conflict : variable indicates whether class is conflict with time constraints
+        conflict = false;
```

Comment:
this variable isn't used because this function only checks unit constraints, not time constraints, right?